### PR TITLE
box_tree: Add viewport_bounds method for clipped bounds queries

### PR DESCRIPTION
**Pull request by jrmoulton**
Add `Tree::viewport_bounds` method that returns the visible area of a node by  intersecting its world bounds with the accumulated clip from all ancestors. This leverages the existing world_clip computation from the ClipBehavior system for efficiency, avoiding tree traversal at query time.

The method respects all ClipBehavior modes (None, PreferLocal, Inherit) and follows the same liveness semantics as other accessor methods. Includes comprehensive tests covering inheritance scenarios, transforms, and edge cases.

Depends on #28

### CODE REVIEW

File: understory_box_tree/src/tree.rs [hunk 1]
```diff
@@ -337,6 +346,26 @@ impl<B: Backend<f64>> Tree<B> {
...
+            return None;
+        }
+        let node = self.nodes.get(id.idx())?.as_ref()?;
+        let unclipped_bounds =
+            transform_rect_bbox(node.world.world_transform, node.local.local_bounds);
+
+        match node.world.world_clip {
+            Some(clip) => Some(unclipped_bounds.intersect(clip)),
+            None => Some(unclipped_bounds),
+        }
```
Comment (tomcur) [inline understory_box_tree/src/tree.rs]: Hmm, this looks very similar to the `world_bounds` calculation. Are there cases where it differs?

The transform is calculated similarly:

https://github.com/endoli/understory/blob/b401400b9b8aa6ed8bd3f5ba55f90411c958788a/understory_box_tree/src/tree.rs#L687-L688

And I believe the same value of `WorldNode::world_clip` is calculated here

https://github.com/endoli/understory/blob/b401400b9b8aa6ed8bd3f5ba55f90411c958788a/understory_box_tree/src/tree.rs#L703-L707

Calling out to `self.world_bounds(id)` also lets tests pass.

Comment (jrmoulton) [reply understory_box_tree/src/tree.rs]: Ah yes looks like it. I assumed the world bounds wasn't respecting the clips
